### PR TITLE
surface 500 and 503 errors to ng:messages.

### PR DESCRIPTION
**Pull request by akash1810**
Further to this, we should be using `ng:messages` more, rather than [`$window.alert`](https://github.com/guardian/grid/blob/master/kahuna/public/js/components/gr-add-label/gr-add-label.js#L15) so that it doesn't block the UI. Though we'd most likely need to give the `ng:message` context so that the message is more meaningful/helpful.

### CODE REVIEW

File: kahuna/public/js/errors/global.html [hunk 1]
```diff
@@ -22,4 +22,14 @@
             to reauthenticate.
         </div>
     </div>
+
+    <div class="global-error warning" ng:message="unknown">
```
Comment (akash1810) [inline kahuna/public/js/errors/global.html]: `warning` class as the error code/severity isn't known.

File: kahuna/public/js/main.js [hunk 2]
```diff
@@ -151,16 +151,29 @@ kahuna.run(['$rootScope', 'mediaApi',
...
-kahuna.factory('httpUnauthorisedInterceptor',
+kahuna.factory('httpErrorInterceptor',
                ['$q', '$rootScope', 'httpErrors',
                 function($q, $rootScope, httpErrors) {
     return {
         responseError: function(response) {
-            if (response.status === httpErrors.unauthorised.errorCode) {
-                $rootScope.$emit('events:error:unauthorised');
+            switch (response.status) {
+                case 0:
```
Comment (theefer) [inline kahuna/public/js/main.js]: When does that happen?

Comment (akash1810) [inline kahuna/public/js/main.js]: I found it happened on POST failure - I changed the [shape of the payload](https://github.com/guardian/grid/blob/master/kahuna/public/js/services/label.js#L25) in the add label message in DEV to test this (`.post({data_: labels})` vs. `.post({data: labels})`). This caused a server error which returned 500 in the network tab, but angular always showed it as 0.

Comment (theefer) [inline kahuna/public/js/main.js]: That's slightly worrying, could we investigate some more? Were there any security errors in the console?

Comment (akash1810) [inline kahuna/public/js/main.js]: qq - would it be better to use `$provide.decorator` like [pandular does](https://github.com/guardian/pandular/blob/master/src/heal.js)?

Comment (theefer) [inline kahuna/public/js/main.js]: Not necessarily, the decorator is useful so we can remap the promise (until after we have tried to reauth) but we don't want to do this here, just spy on errors.

**Comment by theefer** [comment]
:+1:

**Pull request merged**